src/main/java/billgates/usecases/user_join/UserJoinUseCase.java

```diff
@@ -0,0 +1,29 @@
+package billgates.usecases.user_join;
+
+import billgates.database.QueryUserData;
+import billgates.interface_adapters.DatabaseGateway;
+
+import java.util.function.ToDoubleBiFunction;
+
+public class UserJoinUseCase implements UserJoinInputPort{
+    private final DatabaseGateway gateway;
+    private final UserJoinOutputPort outputPort;
+
+    public UserJoinUseCase(DatabaseGateway gateway, UserJoinOutputPort outputPort){
+        this.gateway = gateway;
+        this.outputPort = outputPort;
+    }
+
+    @Override
+    public void join(UserJoinRequestModel model) {
+        QueryUserData userData = gateway.getUserData();
+        if (!userData.getUsers().contains(model.getUsername())){
+            outputPort.display(new UserJoinResponseModel(false, "Not in database"));
```

Comment:
In my opinion, instead of hard coding something like not in the database, you should try some method calls from input.

Reply:
This could refer to one of my concerns: what do we pass in the position of "Not in database"? In other words, which of the instances could provide such key words and pass them here?

Reply:
If the user passed in is not in the database, we should add it to the database and login.